app/src/main/java/com/example/codingquiz/viewmodel/QuestionViewModel.kt

```diff
@@ -14,10 +14,13 @@ class QuestionViewModel(
     private val questionRepository: QuestionRepository,
 ) : ViewModel() {
     private var questions = emptyList<Question>()
-    private var questionIterator = questions.iterator()
+    private var questionIterator = questions.iterator().withIndex()
     private val _question = MutableStateFlow(Question(0, 0, "Questions not loaded yet", emptyList()))
     val question get() = _question.asStateFlow()
 
+    private val _questionNumber = MutableStateFlow(0)
+    val questionNumber = _questionNumber.asStateFlow()
+
     val timeLeft get() = Timer.timeLeft
 
     fun fetchQuestions(categoryId: Int?) {
```

Comment:
```suggestion
    private lateinit var questionIterator: Iterator<IndexedValue<Question>>
    fun fetchQuestions(categoryId: Int) {
        viewModelScope.launch {
            val questions = questionRepository.getRandom(
                quantity = QUESTION_COUNT,
                categoryId = categoryId,
            )

            questionIterator = questions.iterator().withIndex()
            nextQuestion()
        }
    }
```
I checked it and this is enough. You don't have to store `questions` in the field.

Reply:
Done